File: python/p007_10001st_prime.py
#!/usr/bin/python2
#-*- encoding:utf-8 -*-

import itertools

"""
By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we
can see that the 6th prime is 13.

What is the 10,001st prime number?
"""

import euler

def main():
    with euler.TimeWrapper():
        n = 1
        for prime in itertools.count(3, 2):
            if euler.is_prime(prime):
                n += 1
                if n == 10001:
                    break

        print prime

if __name__ == '__main__':
    main()

File: c/p007_10001st_prime.c
#include "euler.h"
#include <stdio.h>
#include <stdlib.h>

/*
 * By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we
 * can see that the 6th prime is 13.
 *
 * What is the 10,001st prime number?
 */

static int run(void) {
  NUM_TYPE result;

  NUM_TYPE n = 1;
  for (result=3;; result+=2) {
    if (is_prime(result)) {
      if (++n >= 10001) {
	break;
      }
    }
  }
  
  printf("%"NUM_TYPE_PRINTF"\n", result);
  return 0;
}

int main(void) {
  int exit_code = time_wrapper(run);
  if (exit_code != 0) {
    perror("error");
  }

  return EXIT_SUCCESS;
}
